app/Http/Controllers/Auth/RegisteredUserController.php

```diff
@@ -21,20 +21,26 @@ class RegisteredUserController extends Controller
      *
      * @return \Illuminate\View\View
      */
-    public function create()
+    public function create(Request $request)
     {
-        
-        $invitationEmail = '';
+        $invitationEmail = $this->getInvitationEmail($request);
 
-        if(request('key')){
-            $invitation = Invitations::where('key',request('key'))
-            ->where('pending',true)
-            ->firstOrFail();
-            
-            $invitationEmail = $invitation->email;
-            
+        return Inertia::render('Auth/Register', [
+            'invitationEmail' => $invitationEmail
+        ]);
+    }
+
+    private function getInvitationEmail(Request $request): string
+    {
+        if (!$request->has('key')) {
+            return '';
```

Comment:
how does this render when the key isn't present?

Reply:
I remember there was a specific reason why I did it like this. It was either debouncing or something like that. Let me go through some scenarios to figure out why.

Reply:
Now I remember. It was to reuse the registration method for both brand new users and people that are invited to be part of a group. If you specify an invite key, like when clicking on the invite link from your email, it will fill out the registration form with the email address that was received. But if you're a brand new user registering, obviously you have no key so it has nothing to fill out.